iroh/src/watchable.rs

```diff
@@ -101,40 +101,62 @@ impl<T: Clone + Eq> Watchable<T> {
     }
 }
 
-/// An observer for a value.
+/// A handle to a value that's represented by one or more underlying [`Watchable`]s.
 ///
-/// The [`Watcher`] can get the current value, and will be notified when the value changes.
-/// Only the most recent value is accessible, and if the thread with the [`Watchable`]
-/// changes the value faster than the thread with the [`Watcher`] can keep up with, then
-/// it'll miss in-between values.
-/// When the thread changing the [`Watchable`] pauses updating, the [`Watcher`] will always
-/// end up reporting the most recent state eventually.
-#[derive(Debug, Clone)]
-pub struct Watcher<T> {
-    epoch: u64,
-    shared: Weak<Shared<T>>,
-}
-
-impl<T: Clone + Eq> Watcher<T> {
-    /// Returns the currently held value.
+/// This handle allows one to observe the latest state and be notified
+/// of any changes to this value.
+pub trait Watcher: Clone {
```

Comment:
I'm not a huge fan of the `DirectWatcher` name.  I wonder if this name can be something else instead?  Maybe `ValueWatcher`?  Or even `WatcherTrait`?  I don't know, just some random names that came out of my keyboard.

Reply:
Yeah, not super happy either.

I *would* like to keep the clean `trait Watcher` name, since I'd rather have people use `impl Watcher` in their APIs than surface `DirectWatcher`... But of course that means no named types :sob:

And I do think `Value` doesn't actually help, since every watcher is a "value" watcher.

`SourceWatcher`?

Not sure.